geode-redis/src/main/java/org/apache/geode/redis/internal/KeyRegistrar.java

```diff
@@ -14,56 +14,147 @@
  */
 package org.apache.geode.redis.internal;
 
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
 import java.util.Map;
 import java.util.Set;
 
+import org.apache.geode.DataSerializer;
+import org.apache.geode.InvalidDeltaException;
 import org.apache.geode.cache.Region;
 
 public class KeyRegistrar {
-  private Region<String, RedisDataType> redisMetaRegion;
+  private Region<ByteArrayWrapper, RedisData> redisDataRegion;
 
-  public KeyRegistrar(Region<String, RedisDataType> redisMetaRegion) {
-    this.redisMetaRegion = redisMetaRegion;
+  public KeyRegistrar(Region<ByteArrayWrapper, RedisData> redisDataRegion) {
+    this.redisDataRegion = redisDataRegion;
   }
 
   /**
    * Checks if the givenKey is associated with the passed data type. If an entry doesn't exist,
    * store the key:datatype association in the metadataRegion
    */
   public void register(ByteArrayWrapper key, RedisDataType type) {
-    RedisDataType existingType = this.redisMetaRegion.putIfAbsent(key.toString(), type);
-    if (!isValidDataType(existingType, type)) {
-      throwDataTypeException(key, existingType);
+    RedisData existingValue = this.redisDataRegion.putIfAbsent(key, transformType(type));
+    if (!isValidDataType(existingValue, type)) {
+      throwDataTypeException(key, existingValue);
+    }
+  }
+
+  /**
+   * This class should go away once all data types implement RedisData.
```

Comment:
Probably a TODO here, so it's more obvious.

Reply:
done